### ISSUE TITLE
Setup Keycloak federation in deployed testbed

### ISSUE THREAD

**Issue by fkr** [SovereignCloudStack/issues#210]
## Definition of Done:

- [ ] Keycloak federation works in testbed 
- [ ] All acceptance criteria are met
- [ ] Changes have been reviewed
- [ ] CI tests have run successfully
- [ ] Documentation has been updated
- [ ] Release Notes have been updated

**Comment by reqa** [comment]
Currently blocked by https://github.com/osism/testbed/issues/1282#issuecomment-1308622772

**Comment by reqa** [comment]
@berendt thinks our hacked approach will be sufficient to continue work and requested to create a [separate issue](https://github.com/osism/testbed/issues/1392). We'll see how we can continue in the hackish way for the mean time. Currently this has repercussions for the openstack v4oidcpassword, so we'll need to debug and work around them too. Pretty annoying to be honest.

**Comment by berendt** [comment]
Isn't it? The certificate works now after redeployment? At least that's what you mentioned in Kurt's issue. We will also just add the missing SN entry to the certificate as you suggested in Kurt's issue (splitter into https://github.com/osism/testbed/issues/1390).

Don't know what's annoying about it if you've found a solution to the problem and it works?

**Comment by reqa** [comment]
The thing that annoyed me is that we are now working again on SSL issues and hacking them locally in the testbed, messing it up as we go. Yes, we have sucessfully experimented with an adjusted certificate, but apparently did not sign the CSR with the correct CA key or so, so as a result nothing works now, neither Web-SSO not `openstack` client. So we debug again now how to more properly hack that. Instead, I would expect that we fix it in the testbed repos and redeploy. That's kind of the approach that I would have expected to take. -- Before I continue digging with rgrep/find/etc to find out where the CA Key is, maybe you know from the top of your head or can give a pointer where to look for it? Somehow we need to get the certificate right in the testbed -- and yes, that is blocking us.

**Comment by berendt** [comment]
https://github.com/osism/testbed/tree/main/contrib/ownca

Mathias will take care of https://github.com/osism/testbed/issues/1390 later.

**Comment by reqa** [comment]
Yeay, https://github.com/osism/testbed/issues/1402 has been resolved!
